pkg/kinflate/resource/util.go

```diff
@@ -90,3 +90,71 @@ func Merge(rcs ...ResourceCollection) (ResourceCollection, error) {
 
 	return all, nil
 }
+
+// Merge will merge all of the entries in the slice of ResourceCollection with Override
+// If there is already an entry with the same GVKN exists, different actions are performed according to value of Behavior field
+// 'create': create a new one;
+// 'replace': replace the data only; keep the labels and annotations
+// 'merge': merge the data; keep the labels and annotations
+func MergeWithOverride(rcs ...ResourceCollection) (ResourceCollection, error) {
+	all := ResourceCollection{}
+
+	for _, rc := range rcs {
+		for gvkn, obj := range rc {
+			if _, found := all[gvkn]; found {
+				switch obj.Behavior {
+				case "", "create":
+					return nil, fmt.Errorf("Create an existing gvkn %#v is not allowed", gvkn)
+				case "replace":
+					originalObj := all[gvkn]
+					obj.Data.SetLabels(mergeMap(originalObj.Data.GetLabels(), obj.Data.GetLabels()))
+					obj.Data.SetAnnotations(mergeMap(originalObj.Data.GetAnnotations(), obj.Data.GetAnnotations()))
+					obj.Data.SetName(originalObj.Data.GetName())
+					all[gvkn] = obj
+				case "merge":
+					originalObj := all[gvkn]
+					obj.Data.SetLabels(mergeMap(originalObj.Data.GetLabels(), obj.Data.GetLabels()))
+					obj.Data.SetAnnotations(mergeMap(originalObj.Data.GetAnnotations(), obj.Data.GetAnnotations()))
+					obj.Data.SetName(originalObj.Data.GetName())
+					mergeData(obj.Data.Object, originalObj.Data.Object, obj.Data.Object)
+					all[gvkn] = obj
+				default:
+					return nil, fmt.Errorf("The behavior of %#v must be one of create, merge and replace", gvkn)
+				}
+			} else {
+				switch obj.Behavior {
+				case "", "create", "replace":
```

Comment:
> Replace non existing configmaps

From the spec doc you wrote, it seems `replace` should also error out.

Reply:
Will fix it.